Libraries/Network/RCTNetworking.mm

```diff
@@ -333,22 +333,60 @@ - (RCTURLRequestCancellationBlock)processDataForHTTPQuery:(nullable NSDictionary
   return callback(nil, nil);
 }
 
-+ (NSString *)decodeTextData:(NSData *)data fromResponse:(NSURLResponse *)response
++ (NSString *)decodeTextData:(NSData *)data fromResponse:(NSURLResponse *)response withCarryStorage:(NSMutableData*)carryStorage
 {
   NSStringEncoding encoding = NSUTF8StringEncoding;
   if (response.textEncodingName) {
     CFStringEncoding cfEncoding = CFStringConvertIANACharSetNameToEncoding((CFStringRef)response.textEncodingName);
     encoding = CFStringConvertEncodingToNSStringEncoding(cfEncoding);
   }
+	
+	BOOL wasCarryStorageProvided = carryStorage != nil;
+	
+	if(carryStorage == nil)
+	{
+		carryStorage = [NSMutableData new];
+	}
```

Comment:
see my comment below about when `carryStorage` wasn't supplied. It's probably cleaner to make a new working variable and keep the original carryStorage

```
NSMutableData *currentCarry = carryStorage ?: [NSMutableData new];
```

then `if (wasCarryStorageProvided)` can be replaced with `if (carryStorage)` and at the end of the function:

```
if (carryStorage != nil) {
  [carryStorage setData:newCarryStorage];
}
```

This way, it's clearer that you're modifying the supplied data